lib/game.rb

```diff
@@ -0,0 +1,83 @@
+class Game
+
+  def initialize(state=Array.new(9, ""))
+    @state = state
+  end
+
+  def set_mark(position, marker)
+    index = position - 1
+    @state[index] = marker
+  end
+
+  def get_mark_at(position)
+    index = position - 1
+    @state[index]
+  end
+
+  def set_marks(row1, row2, row3)
+    0.upto(2) do |i|
+      @state[i] = row1[i]
+      @state[i + 3] = row2[i]
+      @state[i + 6] = row3[i]
+    end
+  end
+
+  def board_as_string
+    string = ''
+    @state.each_with_index do |spot, index|
+      val = (spot != '') ? spot: "-"
+      string += "|" + val.to_s + "|"
+      if index == 2 || index == 5
+        string += "\n"
+      end
+    end
+    string
+  end
+
+  def check_horizontal_win(mark)
+    i = 0
+    while i <= 8
+      if(@state[i] == mark && @state[i+1] == mark && @state[i+2] == mark)
+        return true
+      end
+      i += 3
+    end
+    false
+  end
+
+  def check_vertical_win(mark)
+    0.upto(8) do |i|
```

Comment:
Aren't their only three (3) possible vertical wins?  going up to 8 will reach beyond the size of the array.